src/platform_impl/windows/window.rs

```diff
@@ -125,12 +154,8 @@ impl Window {
     }
 
     pub fn set_title(&self, text: &str) {
-        let text = OsStr::new(text)
-            .encode_wide()
-            .chain(Some(0).into_iter())
-            .collect::<Vec<_>>();
         unsafe {
-            winuser::SetWindowTextW(self.window.0, text.as_ptr() as LPCWSTR);
+            SetWindowTextW(self.hwnd(), OsStr::new(text));
```

Comment:
`IntoParam` is also provided for `&str` (same for the other `OsStr::new()` calls introduced in this PR).